massa-deferred-calls/src/lib.rs

```diff
@@ -0,0 +1,482 @@
+            .expect(MESSAGE_SER_ERROR);
+        let key = deferred_call_prefix_key!(buf_id, slot.to_bytes_key());
+
+        let mut serialized_call: Vec<u8> = Vec::new();
+        for (serialized_key, serialized_value) in self.db.read().prefix_iterator_cf(STATE_CF, &key)
+        {
+            if !serialized_key.starts_with(&key) {
+                break;
+            }
+
+            serialized_call.extend(serialized_value.iter());
+        }
+
+        match self
+            .call_deserializer
+            .deserialize::<DeserializeError>(&serialized_call)
+        {
+            Ok((_rest, call)) => Some(call),
+            Err(_) => None,
+        }
+    }
+
+    /// Returns the total amount of gas booked for a slot
+    pub fn get_slot_gas(&self, slot: &Slot) -> u64 {
+        // By default, if it is absent, it is 0
+        let key = deferred_call_slot_total_gas_key!(slot.to_bytes_key());
+        match self.db.read().get_cf(STATE_CF, key) {
+            Ok(Some(v)) => {
+                let result = self
+                    .call_deserializer
+                    .u64_var_int_deserializer
+                    .deserialize::<DeserializeError>(&v)
+                    .expect(MESSAGE_DESER_ERROR)
+                    .1;
+                result
+            }
+            _ => 0,
+        }
+    }
+
+    /// Returns the base fee for a slot
+    pub fn get_slot_base_fee(&self, slot: &Slot) -> Amount {
+        let key = deferred_call_slot_base_fee_key!(slot.to_bytes_key());
+        match self.db.read().get_cf(STATE_CF, key) {
+            Ok(Some(v)) => {
+                self.call_deserializer
+                    .amount_deserializer
+                    .deserialize::<DeserializeError>(&v)
+                    .expect(MESSAGE_DESER_ERROR)
+                    .1
+            }
+            _ => Amount::zero(),
+        }
+    }
+
+    /// Returns the total amount of gas booked
+    pub fn get_total_gas(&self) -> u128 {
+        match self
+            .db
+            .read()
+            .get_cf(STATE_CF, DEFERRED_CALL_TOTAL_GAS.as_bytes().to_vec())
+            .expect(CRUD_ERROR)
+        {
+            Some(v) => {
+                let result = self
+                    .registry_changes_deserializer
+                    .total_gas_deserializer
+                    .deserialize::<DeserializeError>(&v)
+                    .expect(MESSAGE_DESER_ERROR)
+                    .1;
+                match result {
+                    SetOrKeep::Set(v) => v,
+                    SetOrKeep::Keep => 0,
+                }
+            }
+            None => 0,
+        }
+    }
+
+    pub fn put_entry(
+        &self,
+        slot: &Slot,
+        call_id: &DeferredCallId,
+        call: &DeferredCall,
+        batch: &mut DBBatch,
+    ) {
+        let mut buffer_id = Vec::new();
+        self.call_id_serializer
+            .serialize(call_id, &mut buffer_id)
+            .expect(MESSAGE_SER_ERROR);
```

Comment:
I see you reused the AsyncPool's error messages, which is ok.
But don't hesitate (but not mandatory) to create your own DEFERRED_CALL_ID_SER_ERROR for example.

Reply:
added